features/2025-07-radius-resource-types-contribution.md

```diff
@@ -0,0 +1,292 @@
+                  readOnly: true
+            required:
+                - environment
+    ```
+
+    **Resource Type Schema Enforcement**
+
+    Radius enforces ARM naming conventions for resource type schemas. The following guidelines need to be followed:
+
+    - The `name` field follows the format `Radius.<Category>`, where `<Category>` is a high-level grouping (e.g., Datastores, Messaging, Dapr) For e.g. Radius.Datastores. This is a change from the previous format of `Applications.Datastores` to help users distinguish their own resource types from the Radius Resource types. The core resource type will also need to follow the `Radius.Core` format so that it is consistent across the resource types.
+
+    - The resource type name follows the camel Case convention and is in plural form, such as `redisCaches`, `sqlDatabases`, or `rabbitMQQueues`.
+
+    - Version should be the latest date and follow the format `YYYY-MM-DD-preview`. This is the date on which the contribution is made or when the resource type is tested and validated. For e.g. `2025-07-20-preview`.
+
+    - Properties should follow the camel Case convention and include a description for each property. 
+        - `readOnly:true` set for property automatically populated by Radius Recipes.
+        - `type` could be `integer`, `string` or `object`; Support for `array` and `enum` in progress
+        - `required` for required properties. `environment` should always be a required property.
+
+1. Create Recipes for your resource type in the `recipes` directory. Each Recipe should be organized by cloud provider or technology stack, such as `aws-memorydb`, `azure-redis`, or `kubernetes`. 
+
+    - Each Recipe should include a README.md file that describes how to use the Recipe, including prerequisites, parameters required, and examples.
+    - The Recipes could be a Bicep or Terraform template organized in the respective directories, such as `bicep` or `terraform`.
+    - Contributors are required to add at least one Recipe that works with the resource-type but also encouraged to provide multiple Recipes for different cloud providers or technologies, such as AWS MemoryDB, Azure Redis Cache, or Kubernetes Redis Cache.
+    - Guidelines documentation for Recipes is available [here](https://github.com/Reshrahim/resource-types-contrib/blob/main/contributing-docs/contributing-resource-types-recipes.md#recipes-for-the-resource-type)
+
+1. Manually test the resource type and Recipe locally. Detailed instructions for testing the resource type and Recipe are written [here](https://github.com/Reshrahim/resource-types-contrib/blob/main/contributing-docs/testing-resource-types-recipes.md)
+
+1. Add a functional test for the resource type and Recipe in the `radius` repository to validate the resource type schema and the Recipes provided. The tests will ensure that the resource type can be created, updated, and deleted successfully, and that the Recipes can be deployed without errors.
```

Comment:
Should we specify whether these tests should be locally hosted implementations vs cloud specific (we could leave it open ended but locally hosted would be easiest to test/maintain)